### PR TITLE
derp: add a README.md with some docs

### PR DESCRIPTION
**Pull request by bradfitz**
Updates #docs

### CODE REVIEW

File: derp/README.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+## DERP Regions
+
+Tailscale runs 1 or more DERP nodes (instances of `cmd/derper`) in various
+geographic regions to make sure users have low latency to their DERP home.
+
+Regions generally have multiple nodes per region "meshed" (routing to each
+other) together for redundancy: it allows for cloud failures or upgrades without
+kicking users out to a higher latency region. Instead, clients will reconnect to
+the next node in the region. Each node in the region is required to to be meshed
+with every other node in the region and forward packets to the other nodes in
```
Comment (valscale) [inline derp/README.md:43]: I *think* a connection is only ever forwarded one hop, is that correct?

Comment (bradfitz) [reply derp/README.md:43]: yes

File: derp/README.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+the region. The assumption is that the mesh TCP connections are over a VPC
+that's very fast, low latency, and not charged per byte. The coordination server
+assigns the list of nodes in a region as a function of the tailnet, so all nodes
+within a tailnet should generally be on the same node and not require
+forwarding. Only after a failure do clients of a particular tailnet get split
+between nodes in a region and require inter-node forwarding. But over time it
+balances back out. There's also an admin-only DERP frame type to force close the
```
Comment (valscale) [inline derp/README.md:50]: Would this be a good place to include hints on how to send this frame, how to find connections that are forwarded etc? (The answer might be "finish derp flows.")

Comment (bradfitz) [reply derp/README.md:50]: It's in corp repo: cmd/derpprune.

Comment (bradfitz) [reply derp/README.md]: Done.

File: derp/README.md (hunk 1 above)
Comment (bradfitz) [reply derp/README.md:43]: Done.

**Review by valscale — APPROVED** [review]

**Pull request merged**